api/main_endpoints/routes/Course.js

```diff
@@ -0,0 +1,105 @@
+const express = require('express');
+const router = express.Router();
+const Course = require('../models/Course');
+const {
+  checkIfTokenSent,
+  checkIfTokenValid
+} = require('../util/token-functions');
+const {
+  OK,
+  BAD_REQUEST,
+  UNAUTHORIZED,
+  FORBIDDEN,
+  NOT_FOUND
+} = require('../../util/constants').STATUS_CODES;
+const addErrorLog = require ('../util/logging-helpers');
+
+router.get('/getCourses', (req, res) => {
```

Comment:
There should be a way for the caller to send a a filter in order to get specific items. If there is an empty filter then it will just return all of the items